### PR TITLE
refactor(Core/SpellMgr): simplify dbc data corrections

### PR DESCRIPTION
**Pull request by Kitzunu**
Originally made here https://github.com/TrinityCore/TrinityCore/commit/34b63019ae977ef29606e8b1e0acc2442a63e295
modified and corrected after AC's current spells.

## CHANGES PROPOSED:
- Allow sending error message when we try to modify non-existing spells
- remove the switch case and make it more readable

## TESTS PERFORMED: https://github.com/azerothcore/azerothcore-wotlk/pull/3709#issuecomment-728141638
- Builds
- Tested in-game

## HOW TO TEST THE CHANGES:

1. .cast 9910
2. Fight a mob that is higher level than you
3. check for DBErrors

This is the easiest to test and if this works == Everything works

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Kitzunu** [comment]
These errors are actually Good.

They are DB errors that are due to spells not having correct attrib or whatever to work woth the condition (because I removed the whole switch case)

So when I am done with this, and the DBErrors are gone, it means this works 👍

**Comment by Winfidonarleyan** [comment]
Very very good

**Comment by Kitzunu** [comment]
I'll defenetly look at your commit.

Thought of renaming them too, nice!

**Comment by Kitzunu** [comment]
@Winfidonarleyan perhaps you know why I need to use SpellEntry* spellInfo for the build to work?

SpellInfo* spellInfo doesnt 🤷

**Comment by Winfidonarleyan** [comment]
@Kitzunu 
- https://github.com/TrinityCore/TrinityCore/commit/00f3b01ea72069671b928f8b196a89d4b1a39d55
- https://github.com/TrinityCore/TrinityCore/commit/34b63019ae977ef29606e8b1e0acc2442a63e295

**Comment by Mitradis** [comment]
cant merge after merge "remove ACE Auto_Ptr" and othet astyle:
https://github.com/Mitradis/azerothcore-wotlk/commits/master

**Comment by Mitradis** [comment]
@Winfidonarleyan 
cant merge after merge "remove ACE Auto_Ptr" and othet astyle:
https://github.com/Mitradis/azerothcore-wotlk/commits/master

**Comment by Kitzunu** [comment]
PR DONE!

We can conclude this works because
1. We don't have any DB errors from Conditions or SAI that rely on some spells having a different attribute or something
2. Tested in-game with thorns
3. Builds

**Comment by Winfidonarleyan** [comment]
`SpellEntry` hmm :/ need `SpellInfo`

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by Kitzunu** [comment]
I will change that in the future.
Atm spellentry works

File: src/server/game/Spells/SpellMgr.cpp [hunk 1]
```diff
@@ -3265,3245 +3265,3952 @@ void SpellMgr::LoadSpellCustomAttr()
...
 
-            //////////////////////////////////////////
-            ////////// BREWFEST
-            //////////////////////////////////////////
-            // Apple Trap
-            case 43450:
-                spellInfo->EffectImplicitTargetA[0] = TARGET_UNIT_SRC_AREA_ENEMY;
-                spellInfo->EffectImplicitTargetB[0] = TARGET_UNIT_CASTER;
-                spellInfo->Effect[0] = SPELL_EFFECT_DUMMY;
-
```
Comment (FrancescoBorzi) [inline src/server/game/Spells/SpellMgr.cpp:7212]: missing brackets

Comment (Kitzunu) [reply src/server/game/Spells/SpellMgr.cpp:7212]: that is removed code O.o

Comment (Kitzunu) [reply src/server/game/Spells/SpellMgr.cpp:7212]: Ah I see now...
This was code I actually didn't touch but github diff is bad and does weird shit :D

**Comment by JamesGent** [comment]
.cast 9910
    Fight a mob that is higher level than you
    check for DBErrors

Decided to fight a Fel Reaver at lvl 6 (that went well) and a Stormwind Guard at lvl 74 (they're 75).

No DBE errors on either encounter.

**Pull request merged**